src/fluree/db/ledger/json_ld.cljc

```diff
@@ -138,37 +145,48 @@
   true)
 
 (defn notify
-  [{:keys [conn] :as ledger} {:keys [address t branch] :as commit-notification}]
-  (let [latest-db (ledger-proto/-db ledger {:branch branch})
-        latest-t  (:t latest-db)]
-    ;; note, index updates will have same t value as current one, so still need to check if t = latest-t
-    (cond
-
-      (< t latest-t)
-      (do
-        (log/info "Received update for ledger: " address " at t value: " t
-                  " however, latest-t is more current: " latest-t)
-        false)
-
-      (= t (inc latest-t))
-      (->> (jld-reify/read-commit conn address)
-           <?
-           (jld-reify/merge-commit conn latest-db false)
-           <?
-           (update-local-db ledger))
-
-      ;; missing some updates, dump in-memory ledger forcing a reload
-      (> t (inc latest-t))
-      (do
-        (close-ledger ledger)
-        false))))
+  "Returns false if provided commit update did not result in an update to the ledger because
+  the provided commit was not the next expected commit.
+
+  If commit successful, returns successfully updated db."
+  [{:keys [conn] :as ledger} expanded-commit]
+  (go-try
+    (let [[commit proof] (jld-reify/parse-commit expanded-commit)
+          branch    (keyword (get-first-value expanded-commit const/iri-branch))
+          commit-t  (-> expanded-commit
+                        (get-first const/iri-data)
+                        (get-first-value const/iri-t))
+          latest-db (ledger-proto/-db ledger {:branch branch})
+          latest-t  (- (:t latest-db))]
+      (log/debug "notify of new commit for ledger: " (:alias ledger) " at t value: " commit-t
+                 " where current cached db t value is: " latest-t)
+      ;; note, index updates will have same t value as current one, so still need to check if t = latest-t
+      (cond
+
+        (< commit-t latest-t)
+        (do
+          (log/info "Received commit update for ledger: " (:alias ledger) " at t value: " commit-t
+                    " however, latest-t is more current: " latest-t)
+          false)
+
+        (= commit-t (inc latest-t))
+        (let [updated-db  (<? (jld-reify/merge-commit conn latest-db false [commit proof]))
+              commit-map  (commit-data/json-ld->map commit (select-keys updated-db index/types))
+              updated-db* (assoc updated-db :commit commit-map)]
+          (commit-update ledger branch updated-db* false))
+
+        ;; missing some updates, dump in-memory ledger forcing a reload
+        (> commit-t (inc latest-t))
+        (do
+          (close-ledger ledger)
```

Comment:
_super_ minor, but I think a debug log would be good here since we're changing internal ledger state. it might make it easier to figure stuff out if we had to, well, debug an issue later.